### PR TITLE
fix(sync): route explicit selected catch-up through RFC-64 lane

### PR DESCRIPTION
**Pull request by branarakic**
## Impact

An Edge node that explicitly subscribes to a selected public context graph now uses the RFC-64 selected-SWM completion lane for operators declared as complete SWM providers. Other peers continue through the existing ordinary catch-up path.

Previously, the subscribe worker correctly resolved and ranked complete providers, but discarded that authority classification when it crossed the worker/daemon RPC boundary. The receiver therefore used generic SWM sync and could not produce authority-bound coverage or selected-lane completion evidence.

## Before

```mermaid
sequenceDiagram
    participant Edge as Edge subscriber
    participant Worker as Catch-up worker
    participant Daemon as DKG daemon
    participant Provider as Complete SWM provider

    Edge->>Worker: Subscribe with SWM enabled
    Worker->>Worker: Resolve and rank complete provider
    Worker->>Daemon: syncSharedMemory
    Daemon->>Provider: Generic SWM sync
    Provider-->>Daemon: Data without selected-lane result
    Daemon-->>Worker: Generic result
```

## After

```mermaid
sequenceDiagram
    participant Edge as Edge subscriber
    participant Worker as Catch-up worker
    participant Daemon as DKG daemon
    participant Provider as Complete SWM provider

    Edge->>Worker: Subscribe with SWM enabled
    Worker->>Worker: Resolve and rank complete provider
    Worker->>Daemon: syncSelectedSharedMemory
    Daemon->>Provider: RFC-64 selected SWM sync
    Provider-->>Daemon: Data plus authoritative coverage
    Daemon-->>Worker: Historical shared result shape
```

## Implementation

- Route only peers in `authoritativeSharedMemoryPeerIds` through a new worker RPC method.
- Bridge that RPC to `syncSelectedSharedMemoryFromPeerDetailed` with foreground priority and selected-SWM admission.
- Unwrap the typed selected result at the bridge so the worker protocol remains backward-compatible.
- Preserve the generic sync method for fallback and non-authoritative peers.

### Local tests and builds

- `catchup-runner-worker-impl.test.ts`
- `catchup-runner-worker-lifecycle.test.ts`
- 59/59 focused tests passed.
- Agent build, type/package-root checks, and CLI build passed.
- `git diff --check` passed.

### Testnet hotfix validation

Exact source commit: `67a3ada66fbc888e0b995ff437c84aacf6bc3077`.

A fresh disposable Testnet Edge receiver explicitly subscribed to registered public CG `bb-open-20260806-a7190`. Automatic on-connect, reconciler, and generic shared-memory-on-connect recovery were disabled. The configured complete provider was the existing publisher Edge peer.

- Publisher SWM: 32 assets, 9,868 triples.
- Cold receiver SWM: 32 assets, 9,868 triples.
- Exact content digest on both nodes: `sha256:5d6fc2ed4cf90d53eb32834380fcf2f8e325f06483e69d81921a3251987aadaf`.
- Coverage: 32/32, `manifestComplete=true`, `descriptorsAuthoritative=true`, `materializationFailures=0`, `fromAuthority=true`.
- Selected continuation stopped `plane-proven` after one pass.
- Transport was ordinary DKG relay; no Tailscale data path was used.

The aggregate catch-up job ended `unreachable` because this corpus has no VM content and the durable responder classified one metadata-only response as incomplete. That is recorded as a separate VM/aggregate-status follow-up; the SWM plane itself completed with exact parity and authoritative terminal evidence.

## Risk

The behavior change is narrow and opt-in: it applies only when RFC-64 activation declares a peer as a complete SWM provider for the selected CG. Fallback peers and graphs without this configuration retain the existing path.

### CODE REVIEW

**Comment by branarakic** [comment]
Additional Testnet restart-persistence evidence for exact head `67a3ada66fbc888e0b995ff437c84aacf6bc3077`:

- gracefully stopped and restarted the cold receiver from its preserved store
- receiver identity remained `12D3KooWKtiYucfF51Se1qYzBKDnzUPeNpi27Yc1EvF1Pin76g5V`
- post-restart SWM inventory remained **32 graph-scoped assets / 9,868 triples**
- a fresh canonical `?g ?s ?p ?o` comparison across the publisher and receiver produced the same digest on both: `sha256:1301424dc4d410728d1267025896fea821f92fc348d3acc2855dfee67a7678d7`

The data path remained ordinary DKG Testnet transport. This confirms that the selected-lane materialization is durable across process restart.

File: packages/cli/src/catchup-runner.ts [hunk 1]
```diff
@@ -1164,6 +1164,27 @@ class WorkerCatchupRunner implements CatchupRunner {
...
+              typeof source === 'string' ? source : undefined,
+            ),
+            selectedSwmPriority: true,
+          },
+        );
+        // The worker already owns the authority verdict and attaches
+        // `fromAuthority` to the returned coverage. Keep the RPC payload on the
+        // historical SharedMemorySyncResult shape while selecting the typed
+        // RFC-64 producer here, at the in-process agent boundary.
+        return selected.shared;
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner.ts]: 🔴 Bug: Selected SWM completion verdict is dropped before the worker can use it

**What's wrong**
The selected SWM API intentionally separates terminal completion from raw diagnostic counters. By returning only selected.shared, the worker falls back to the old raw-counter completion policy. That can mark a successfully completed selected-provider catchup as failed or authority-unanswered whenever the selected lane resolved an earlier voluntary yield but kept the original failure counters for observability.

**Example**
A complete RFC-64 provider starts with a snapshot yield: shared.failedPhases=1, snapshotPlaneIncomplete=1. The selected continuation then materializes the missing snapshots, so syncSelectedSharedMemoryFromPeerDetailed returns selectedScopeComplete=true and shared.resolvedSnapshotPlaneIncomplete=1, while preserving failedPhases=1 for diagnostics. This bridge returns only shared; the worker sees failedPhases>0 and treats the authority round as not clean, so SWM proof/readiness can fail even though the selected lane reached its terminal boundary.

**Suggested direction**
Return a discriminated selected result through the bridge, or otherwise normalize/classify the selected shared-memory payload so the worker uses selectedScopeComplete/resolved freshness when deciding clean completion and authority proof.

**For Agents**
Look at the new syncSelectedSharedMemory bridge in packages/cli/src/catchup-runner.ts and the worker result type in packages/cli/src/catchup-runner-worker-impl.ts. Preserve raw diagnostics, but carry the selected lane terminal verdict through the RPC or classify selected shared-memory results with the freshness/selected completion semantics. Add a worker-level test where selectedScopeComplete=true with raw resolved snapshot/metadata yield counters still present proves shared-memory readiness and does not set authorityUnanswered.

File: packages/cli/src/catchup-runner-worker-impl.ts [hunk 2]
```diff
@@ -389,7 +389,15 @@ async function runCatchup(request: CatchupRunRequest): Promise<CatchupJobResult>
     const syncSharedMemory = (
       { priority, source }: CatchupPlaneContext,
     ): Promise<CatchupSharedMemoryResult> =>
-      invoke<CatchupSharedMemoryResult>('syncSharedMemory', peerId, request.contextGraphId, priority, source)
+      invoke<CatchupSharedMemoryResult>(
+        authoritativeSharedMemoryPeerIds.has(peerId)
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner-worker-impl.ts]: 🟡 Issue: Avoid adding a second string RPC lane just to select the SWM producer

**What's wrong**
This adds another ad-hoc method name to an already untyped worker bridge, duplicates the ordinary SWM bridge logic, and then flattens the selected typed result back to `selected.shared`. The structure increases coupling between `prepareCatchup`, the worker's authority set, and the parent switch without giving the worker a clearer model. The selected API boundary would be cleaner if the bridge owned the producer selection in one place instead of spreading it across a conditional method string and a near-copy switch case.

**Example**
The ordinary and selected bridge paths both unpack `[peerId, contextGraphId, priority, source]` and build the same admission options. If another admission option or validation rule is added later, it now has to be updated in both `syncSharedMemory` and `syncSelectedSharedMemory`, while the worker also has to keep selecting the matching magic method string.

**Suggested direction**
Make the selected-vs-ordinary choice an explicit parameter of a single shared-memory bridge operation, or hide the selected agent API behind one `syncSharedMemoryForCatchup` helper in the parent. That deletes the duplicated bridge case and keeps the worker from knowing parent-side method names.

**For Agents**
In `packages/cli/src/catchup-runner.ts`, factor the shared argument parsing/admission-options construction and collapse the two RPC branches into one explicit shared-memory request shape, for example by passing a `selected` flag/mode from `catchup-runner-worker-impl.ts`. Preserve the existing behavior: selected providers still call `syncSelectedSharedMemoryFromPeerDetailed` with `selectedSwmPriority: true`, ordinary peers still call `syncSharedMemoryFromPeerDetailed`, and the worker receives the same shared-memory result shape unless you intentionally thread the selected result through. Cover both ordinary and selected-provider bridge paths.

File: packages/cli/src/catchup-runner.ts [hunk 3]
```diff
@@ -1164,6 +1164,27 @@ class WorkerCatchupRunner implements CatchupRunner {
...
+      case 'syncSelectedSharedMemory': {
+        const [peerId, contextGraphId, priority, source] = args as [
+          string, string, number | undefined, unknown,
+        ];
+        const selected = await agent.syncSelectedSharedMemoryFromPeerDetailed(
+          peerId,
+          [contextGraphId],
+          {
+            ...(priority === undefined ? {} : { priority }),
+            source: normalizeSyncAdmissionSource(
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner.ts]: 🟡 Issue: Selected-SWM source clamping is not covered by the bridge tests

**What's wrong**
This change adds a third worker RPC route that crosses the same structured-clone boundary as the existing sync routes. The production code clamps `source`, but the tests only prove clamping for the old routes. Because the new test uses a valid source string, it would not catch a regression that leaked arbitrary source values into selected-SWM scheduler diagnostics.

**Example**
Failing-test sketch: invoke `syncSelectedSharedMemory` with `['peer-swm', 'cg-x', 2000, 'durable:urn:cg:private:abc']` or a non-string source, then assert `calls.selectedShared[0].at(-1).source === 'unspecified'`.

**Suggested direction**
Extend the untrusted-edge source normalization test to cover the new `syncSelectedSharedMemory` RPC path, including an invalid or identifier-bearing source value.

**For Agents**
Look in `packages/cli/test/catchup-runner-worker-lifecycle.test.ts` near the bridge source-clamping test. Add `syncSelectedSharedMemory` to that regression coverage and preserve the current assertions that valid selected SWM calls still include `selectedSwmPriority: true` and return `selected.shared`.

File: packages/cli/src/catchup-runner.ts (hunk 1 above)
Comment (branarakic) [reply packages/cli/src/catchup-runner.ts]: Addressed in 541b913ed. The bridge now returns the discriminated selected result rather than flattening it. The worker classifies its full shared diagnostics with classifySharedMemoryFreshness plus selectedScopeComplete, then uses that lane-specific verdict for coverage tracking, completion evidence, authorityAnswered, and peer success. The worker regression now has selectedScopeComplete=true while raw failedPhases/snapshotPlaneIncomplete remain 1 and are superseded by resolvedSnapshotPlaneIncomplete=1; it proves verified readiness and authorityUnanswered=false.

File: packages/cli/src/catchup-runner-worker-impl.ts (hunk 2 above)
Comment (branarakic) [reply packages/cli/src/catchup-runner-worker-impl.ts]: Addressed in 541b913ed. There is now one syncSharedMemory worker RPC operation. Its explicit final boolean selects ordinary versus RFC-64 selected production inside the parent bridge, where shared admission parsing and source normalization live once. Only literal true enters the selected lane; ordinary callers retain the existing producer and result.

File: packages/cli/src/catchup-runner.ts (hunk 3 above)
Comment (branarakic) [reply packages/cli/src/catchup-runner.ts]: Addressed in 541b913ed. The untrusted-edge regression now invokes the same shared-memory bridge in selected mode with an identifier-bearing source and verifies source=unspecified together with selectedSwmPriority=true. The ordinary non-string source case remains covered as well.

Comment (otReviewAgent) [reply packages/cli/src/catchup-runner.ts]: 🟡 Issue: Avoid overloading the worker RPC with a trailing boolean selector

**What's wrong**
The bridge now hides two different agent operations behind one method string and a positional `unknown` boolean. This is a brittle boundary: the selected lane is not represented in the command name or argument shape, and the long comment is compensating for an implicit contract. As this bridge grows, that kind of hidden mode flag is exactly how unrelated call sites become coupled to feature-specific routing.

**Example**
The ordinary and selected bridge calls differ only by a fifth positional value: `['peer-swm', 'cg-x', 2000, 'catchup-foreground', true]`. That is easy to omit or reorder because the method name still says only `syncSharedMemory`.

**Suggested direction**
Use an explicit bridge operation or structured discriminated payload for the selected lane. That would make the two agent methods visible in the RPC contract and remove the `unknown` trailing flag plus the fail-open positional convention.

**For Agents**
Look at `WorkerCatchupRunner`'s `syncSharedMemory` bridge case and the worker implementation call site. Preserve ordinary routing for normal peers and `selectedSwmPriority: true` for RFC-64 selected providers, but split this into an explicit RPC method such as `syncSelectedSharedMemory` or a discriminated command object. Tests should assert explicit selected routing while ordinary `syncSharedMemory` stays ordinary.

File: packages/cli/src/catchup-runner-worker-impl.ts [hunk 4]
```diff
@@ -568,13 +623,14 @@ async function runCatchup(request: CatchupRunRequest): Promise<CatchupJobResult>
       // Shared memory carries no verified-private-only signal, so the shared
       // evidence only ever has data/empty set — the same reducer still applies.
       const sharedEvidence = catchupPeerPlaneEvidence(shared, {
+        completedWithoutFailure: sharedCompletedWithoutFailure,
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner-worker-impl.ts:645]: 🔴 Bug: Resolved selected-SWM empty completions can still be reported unready

**What's wrong**
The change correctly uses `selectedScopeComplete` to classify the selected peer as clean, but only for per-peer evidence. For selected SWM results that complete without inserting data, the only readiness proof is the empty/no-new-data path, and that path still consults raw aggregate failure counters. Because selected diagnostics intentionally retain resolved voluntary yields, a successful selected completion can be downgraded to `unreachable`.

**Example**
A selected provider can return `selectedScopeComplete: true` with `shared: { completedPhases: 2, insertedDataTriples: 0, emptyResponses: 1, timedOutPhases: 1, metadataContinuationYields: 1, resolvedMetadataContinuationYields: 1, failedPeers: 0, failedPhases: 0 }`. `classifySharedMemoryFreshness` treats this as clean, so line 626 records a clean empty peer. But readiness later sees raw `timedOutPhases: 1`, rejects the unanimous-empty proof, and reports shared memory unready even though the selected lane completed.

**Suggested direction**
Carry the selected freshness verdict through to the final plane readiness decision, or normalize a readiness-only diagnostic view before applying empty/no-new-data proof rules.

**Confidence note**
The exact producer shape should be verified with an end-to-end selected-SWM empty/no-new-insert continuation case, but the reducer/readiness predicates show the mismatch directly.

**For Agents**
Look at `packages/cli/src/catchup-runner-worker-impl.ts` around selected SWM accumulation and `packages/cli/src/context-graph-readiness.ts` / `catchupPlaneProvenByUnanimousEmpty`. Preserve raw diagnostics for telemetry, but make the readiness verdict consume the selected freshness normalization or add explicit selected-complete SWM proof. Add a case where a selected complete provider has no new inserted data and only resolved yield counters, and prove the catch-up marks shared memory ready.

File: packages/cli/src/catchup-runner-worker-impl.ts [hunk 5]
```diff
@@ -77,8 +79,36 @@ parentPort!.on('message', async (message: any) => {
...
+ * provider returns its discriminated terminal verdict as well, so the Worker
+ * never has to infer completion from diagnostic counters that deliberately
+ * retain resolved voluntary yields.
+ */
+type CatchupSelectedSharedMemoryResult = SelectedSharedMemorySyncResult & {
+  /** Projected for the generic catch-up admission retry policy. */
+  readonly deferredBackpressure?: number;
+};
+
+type CatchupSharedMemoryResult = SharedMemorySyncResult | CatchupSelectedSharedMemoryResult;
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner-worker-impl.ts]: 🟡 Issue: Normalize selected shared-memory results before they enter generic reducers

**What's wrong**
This change lets the selected-SWM wrapper leak through the worker’s generic catch-up reduction path. The result is a mode-aware union plus casts and repeated special cases in pass tracking, evidence, and success accounting. That makes the selected lane a cross-cutting concern instead of a local boundary adaptation, so future changes must remember several coordinated branches to preserve the same semantics.

**Example**
A selected result must go through three coordinated branches: compute `sharedCompletedWithoutFailure`, pass it into evidence, then pass `selectedScopeComplete` into success accounting. Missing any one of those sites falls back to the raw durable-style classifier even though selected SWM deliberately retains superseded failure counters.

**Suggested direction**
Introduce one internal normalized shared-memory plane representation or a single `classifyCatchupSharedMemoryResult` helper, and remove selected-specific optional booleans from the generic reducers. The selected lane should be handled at the boundary, not re-discovered throughout accumulation.

**For Agents**
Look at `syncSharedMemory`/`accumulate` in `catchup-runner-worker-impl.ts` and the helper signatures in `catchup-runner.ts`. Preserve selected-SWM diagnostics and terminal-completion semantics, but unwrap/classify once at the worker boundary into a normalized shared-plane model such as `{ payload, completedWithoutFailure }`, then have evidence, pass tracking, and peer success consume that normalized shape. Tests should prove selected SWM with retained resolved failures still counts as clean only when `selectedScopeComplete` is true.

File: packages/cli/src/catchup-runner-worker-impl.ts [hunk 6]
```diff
@@ -453,18 +499,27 @@ async function runCatchup(request: CatchupRunRequest): Promise<CatchupJobResult>
...
     }: PeerRound,
     isContinuationRound = false,
   ): void => {
     let peerDenied = false;
+    const selectedShared = selectedSharedMemoryResult(sharedResult);
+    const shared = sharedMemoryPayload(sharedResult);
+    const sharedCompletedWithoutFailure = shared
+      ? selectedShared
+        ? classifySharedMemoryFreshness(shared, {
+          complete: selectedShared.selectedScopeComplete,
```
Comment (otReviewAgent) [inline packages/cli/src/catchup-runner-worker-impl.ts]: 🟡 Issue: Selected SWM has no fail-closed worker test

**What's wrong**
The behavior being changed depends on `selectedScopeComplete` overriding otherwise clean-looking shared-memory diagnostics. Without a regression test for `false`, the suite can stay green if the worker later falls back to raw durable-style classification or drops the selected completion argument, which would give false confidence in catch-up success and plane-completion diagnostics.

**Example**
A selected SWM provider returns `{ kind: 'selected-shared-memory', shared: { insertedDataTriples: 7, completedPhases: 1, failedPhases: 0 }, selectedScopeComplete: false }`. The worker should not add this to `cleanPlaneCompletions.sharedMemory.verifiedDataPeers`, should not mark the shared-memory authority answered cleanly, and should not count that peer as succeeded solely from the selected SWM lane.

**Suggested direction**
Cover the negative selected-scope path at the worker aggregation boundary, not only the happy path where recovered counters make the selected result clean.

**For Agents**
Add a focused catch-up worker test near `packages/cli/test/catchup-runner-worker-impl.test.ts` for an authoritative RFC-64 SWM provider returning `selectedScopeComplete: false` with otherwise clean/data diagnostics. Assert the selected lane is still invoked, but clean shared-memory evidence and `peersSucceeded` do not treat it as a clean completion.

File: packages/cli/src/catchup-runner-worker-impl.ts (hunk 5 above)
Comment (branarakic) [reply packages/cli/src/catchup-runner-worker-impl.ts]: Fixed in c89f04f42. The RPC wire union now ends at , producing one internal  model. Pass tracking, evidence, authority accounting, and peer success consume that normalized model; the generic helper no longer accepts a selected-specific boolean.

File: packages/cli/src/catchup-runner-worker-impl.ts (hunk 6 above)
Comment (branarakic) [reply packages/cli/src/catchup-runner-worker-impl.ts]: Fixed in c89f04f42. Added a worker-boundary regression for an authoritative selected provider returning clean data diagnostics with . It asserts the selected lane was invoked, no verified/selected-complete SWM evidence is emitted, the authority remains unanswered, and only the separate durable curator counts as succeeded.

File: packages/cli/src/catchup-runner-worker-impl.ts (hunk 5 above)
Comment (otReviewAgent) [reply packages/cli/src/catchup-runner-worker-impl.ts]: 🟡 Issue: Selected-SWM proof is threaded as an evidence side-channel

**What's wrong**
The PR adds a feature-specific optional counter and then patches multiple generic reducers/predicates to recognize it. That makes the selected-SWM terminal verdict easy to drop and makes future proof types likely to spread through the same surfaces again.

**Example**
A future runner that calls `catchupPeerPlaneEvidence(shared, { plane: 'shared-memory', completedWithoutFailure: true })` for a selected-SWM result will get no selected-scope proof unless it also knows to repeat the line-649 mutation. That contradicts the reducer's stated role as the single definition of one peer's evidence.

**Suggested direction**
Collapse this into the evidence model. For example, have the reducer accept a typed lane/proof verdict and emit a unified positive proof counter or proof-kind field, then expose one canonical `catchupPlaneProvenPositively` predicate. Keep `verifiedDataPeers` as telemetry if needed, but do not require each caller to manually OR selected-scope proof into every readiness gate.

**For Agents**
Look at `catchupPeerPlaneEvidence`, `CatchupPlaneCompletionEvidence`, `catchupPlaneReady`, `catchupPlaneReadinessThisRun`, and the worker `accumulate` path. Preserve separate transfer telemetry for inserted/verified data, but make selected-SWM terminal proof part of the canonical evidence reduction rather than a post-reducer mutation. Add/keep tests proving selected-SWM zero-insert terminal proof settles readiness and that incomplete selected scope remains unresolved.

File: packages/cli/src/catchup-runner.ts (hunk 3 above)
Comment (otReviewAgent) [reply packages/cli/src/catchup-runner.ts]: 🟡 Issue: Avoid hiding selected-lane routing behind a positional boolean RPC flag

**What's wrong**
This adds a mode flag to an already loosely typed worker bridge, increasing positional coupling between caller and bridge. The implementation works, but the structure is brittle and makes selected provider routing look like an incidental extra argument rather than a distinct lane with its own result contract.

**Example**
The selected call shape is now `invoke('syncSharedMemory', peerId, contextGraphId, priority, source, true)`. A caller that omits or misorders the fifth argument still calls a valid method and silently routes through ordinary shared memory, which is hard to see at the call site because the mode is just a trailing boolean in an untyped args array.

**Suggested direction**
Give the selected lane an explicit boundary: a dedicated RPC method is the simplest shape, or a typed/discriminated payload if the method must stay shared. That removes the boolean mode branch and makes the bridge contract self-describing.

**For Agents**
Look at the worker `invoke('syncSharedMemory', ...)` call in `catchup-runner-worker-impl.ts` and the bridge dispatch in `WorkerCatchupRunner`. Preserve ordinary shared-memory sync and selected-SWM sync behavior, but replace the trailing boolean with either a separate `syncSelectedSharedMemory` RPC method or a discriminated request object. Keep lifecycle tests for both routes.

File: packages/cli/src/catchup-runner-worker-impl.ts (hunk 6 above)
Comment (otReviewAgent) [reply packages/cli/src/catchup-runner-worker-impl.ts]: 🟡 Issue: The clean-progress gate for selected SWM is not directly covered

**What's wrong**
The PR adds a second guard that prevents an RFC-64 selected provider from proving readiness when its diagnostics still classify as failed. Current tests prove the terminal flag can promote resolved historical yields, and that `selectedScopeComplete: false` fails closed, but they would still pass if the `progress.completedWithoutFailure` conjunct were accidentally dropped. That leaves a risky readiness boundary without direct validation.

**Example**
A regression test could return `{ kind: 'selected-shared-memory', selectedScopeComplete: true, shared: { ...sharedResult(), failedPhases: 1 } }` for the authoritative SWM peer and assert `selectedScopeCompletePeers` stays `0`, `authorityUnanswered` remains true, and the walk does not settle shared memory from that peer.

**Suggested direction**
Add a focused regression test for a malformed or inconsistent selected-SWM result where the terminal flag is true but the progress classifier is not clean, so this guard cannot be removed without a red test.

**For Agents**
Add a worker catch-up test near `packages/cli/test/catchup-runner-worker-impl.test.ts` that exercises an authoritative selected SWM response with `selectedScopeComplete: true` but unresolved failure/timed-out diagnostics. Preserve the behavior that the selected terminal flag is not enough unless `classifySharedMemoryFreshness` reports clean completion.
